### ISSUE TITLE
Make Mailu's templates more flexible and Ansible-friendly

### ISSUE THREAD

**Issue by Ezwen** [Mailu/Mailu#2519]
Internally, Mailu relies on several Jinja2 templates to generate the configuration files for setting up or upgrading a Mailu installation. For example, for a 1.9 setup based on Docker compose, the used templates are:

- https://github.com/Mailu/Mailu/blob/master/setup/flavors/compose/docker-compose.yml
- https://github.com/Mailu/Mailu/blob/master/setup/flavors/compose/mailu.env

These templates are in principle only designed to be used by the [Mailu setup utility](https://setup.mailu.io/), a form provided to users to configure the generation.

In my case, to facilitate deployments and upgrades, I decided to write [a basic Ansible role](https://git.mandragot.org/ezwen-ansible/mailu) that relies on the very same Jinja2 templates to generate configuration files and setup a compose-based Mailu instance. This avoids writing and maintaining templates solely for the Ansible role, with the risk they become outdated with next Mailu versions. Also my goal is to never have to manually edit the compose file or the env file, once generated.

I also aim (a) to use Mailu behind a reverse proxy, and (b) to provide the certificates to Mailu myself. I manage these points with [a template to generate a second compose file to be loaded atop the Mailu one](https://git.mandragot.org/ezwen-ansible/mailu/src/main/templates/docker-compose-extras.yml.j2) (using multiple `-f` compose flags).

----

However, the Jinja2 templates used in Mailu are not directly usable with the use case I describe above:

1. The Mailu template for the compose file will always generate lines to publish ports 80 and 443, with no way to control whether these are generated or not. 
2. In the Mailu template for the env file,  a function call `secret(16)` is used to generate the secret key. This function is not known by Ansible.
3. In the Mailu template for the env file, the default timezone value `Etc/UTC` is hardcoded.
4. In the Mailu template for the env file, boolean values such as `admin_enabled` are directly printed in the produced document. However, Ansible will by default print `True` or `False` instead of `true` or `false`.
5. Ansible by default enables Jinja2 block trimming, meaning the first newline after a block is deleted automatically. Yet, Mailu's Jinja2 blocks are almost always indented, ie. there are space characters between the beginning of the line and the beginning of the block. Hence when block trimming occurs, extra spaces remain and wreaks the YAML structure. 

For my role, my solutions were:
- For points 1-4, I wrote [a script](https://git.mandragot.org/ezwen-ansible/mailu/src/main/templates/bump-templates.sh) that will download the required versions of Mailu's compose templates, and modify them as required (remove ports 80 and 443, replace the function call by a variable, safely cast booleans to lowercase strings, etc.)
- For point 5, I simply disabled Ansible block trimming (`trim_blocks: no`).

---

My question now is, if I am not the only one in that case: **would it be interesting to make Mailu's templates more flexible and Ansible-friendly "out of the box"?**

This would only require small changes:
- Make ports 80 and 443 only published if a given variable is true.
- Provide the secret key to the template using a variable (and generate it before calling the template)
- Provide the timezone to the template using a variable.
- Make sure to have all printed boolean values go through a `| string | lower` pipe for safety.
- Optionally, remove indentation on Jinja2 block declarations.

Sorry for the long post, turns out explaining my case was more verbose than I expected :sweat_smile:

**Comment by ghostwheel42** [comment]
Sorry for not answering promptly.
Sure, you can submit a PR to change the templates while keeping the current functionality. 
Some settings have a fixed value intentionally. TZ for example should always be left at UTC to make debugging the Mail flow easier.

**Comment by Ezwen** [comment]
> Sorry for not answering promptly.

No worries, not in a hurry :)

> Sure, you can submit a PR to change the templates while keeping the current functionality.

Thanks! I'll try to look into that.

> Some settings have a fixed value intentionally. TZ for example should always be left at UTC to make debugging the Mail flow easier.

I see. Would it be OK still to set this value with a Jinja variable in the template, and to set the fixed value of this variable as `UTC`  when Mailu calls the template?

**Comment by ghostwheel42** [comment]
> Would it be OK still to set this value with a Jinja variable in the template, and to set the fixed value of this variable as `UTC` when Mailu calls the template?

Yes. That would be okay. We just do not want to encourage changing this value.
